### ISSUE TITLE
Add Julia Linter

### ISSUE THREAD

**Issue by JSybrandt** [w0rp/ale#627]

**Comment by MagB93** [comment]
Maybe an idea would be to use the interface from Julia for Neovim [Neovim.jl](https://github.com/bfredl/Neovim.jl) and then compute the Linting with [Lint.jl](https://github.com/tonyhffong/Lint.jl) and then pass is back? The package provides Linting for files and even packages. Otherwise I'd like to help, however I'm not that good in Vimscript.

**Comment by w0rp** [comment]
Does it work in Vim 8? I'll only consider working with plugins which support both Vim 8 and NeoVim.

**Comment by MagB93** [comment]
As far as I can tell sadly no. But I went through their feature list and it appears that Lint.jl can be used to start a [LintServer]( https://github.com/tonyhffong/Lint.jl/blob/master/docs/features.md#lintserver). The [Atom-Linter](https://github.com/TeroFrondelius/linter-julia) for Julia is using that Server btw.

**Comment by w0rp** [comment]
https://github.com/JuliaEditorSupport/LanguageServer.jl might be a better bet, as I've already implemented support for Language Server Protocol via stdio.

**Comment by MagB93** [comment]
Yes that's true! I think in the end both paths should deliver the same results.

**Comment by bstellato** [comment]
I would also appreciate this feature. I am not familiar with how ale works, but is there anything that could be done similar to [syntastic julia](https://github.com/vim-syntastic/syntastic/blob/b9d35359bef21268240b074c0d15d57d71642249/syntax_checkers/julia/lint.vim)?

**Comment by w0rp** [comment]
It probably isn't too hard to do something similar to what was done for Syntastic. The way that was written, it reminds me of the R linter.

I'm not familiar with Julia myself. Someone who is can give this a go.

**Comment by m-wells** [comment]
I realize I'm a little late to the party here but I came across an [Asynchronous vim lint checker for julia code](https://github.com/AtsushiSakai/julia.vim).  It supports Vim8 and uses [Lint.jl](https://github.com/tonyhffong/Lint.jl).

**Comment by w0rp** [comment]
If someone wants to add support for `Lint.jl`, go ahead and create a pull request.

**Comment by IngoHeimbach** [comment]
Look at https://github.com/IngoHeimbach/ale/blob/master/ale_linters/julia/lintjl.vim for an ale Lint.jl integration that is quite similar to the Syntastic implementation. It works, but it is very slow since the Lint.jl package components are compiled each time a new Julia process is started. I think using the linting server is the way to go.

**Comment by IngoHeimbach** [comment]
There is another alternative: Lint.jl can be precompiled with [PackageCompiler.jl](https://github.com/JuliaLang/PackageCompiler.jl). Afterwards, running new Julia process for Lint.jl is very fast.
